File: process-assets.js
import * as Jimp from 'jimp';

async function processImages() {
  console.log('Processing Ninja Sprite...');
  const ninjaPath = 'public/assets/sprites/ninja.png';
  const ninja = await Jimp.read(ninjaPath);
  
  // Make white background transparent
  ninja.scan(0, 0, ninja.bitmap.width, ninja.bitmap.height, function(x, y, idx) {
    const r = this.bitmap.data[idx + 0];
    const g = this.bitmap.data[idx + 1];
    const b = this.bitmap.data[idx + 2];
    
    // If it's pure white or very close to white (background of DALL-E)
    if (r > 240 && g > 240 && b > 240) {
      this.bitmap.data[idx + 3] = 0; // Alpha to 0
    }
  });
  await ninja.writeAsync(ninjaPath);
  console.log('Ninja background removed!');

  const backgrounds = ['bg_forest.png', 'bg_beach.png', 'bg_castle.png'];
  for (const bg of backgrounds) {
    console.log(`Cropping ${bg}...`);
    const imgPath = `public/assets/backgrounds/${bg}`;
    try {
      const image = await Jimp.read(imgPath);
      // Crop top half
      image.crop(0, 0, image.bitmap.width, Math.floor(image.bitmap.height / 2));
      await image.writeAsync(imgPath);
      console.log(`Cropped ${bg} successfully!`);
    } catch(e) {
      console.log(`Could not process ${bg}:`, e.message);
    }
  }
}

processImages().catch(console.error);

File: process_assets.py
import os
from PIL import Image

def process_ninja():
    print("Processing Ninja Sprite...")
    path = 'public/assets/sprites/ninja.png'
    if not os.path.exists(path):
        print("Ninja sprite not found!")
        return
        
    img = Image.open(path).convert("RGBA")
    datas = img.getdata()
    
    newData = []
    for item in datas:
        # Change all white (also shades of whites)
        # to transparent
        if item[0] > 240 and item[1] > 240 and item[2] > 240:
            newData.append((255, 255, 255, 0))
        else:
            newData.append(item)
            
    img.putdata(newData)
    img.save(path, "PNG")
    print("Ninja background removed!")

def process_backgrounds():
    backgrounds = ['bg_forest.png', 'bg_beach.png', 'bg_castle.png']
    for bg in backgrounds:
        path = f'public/assets/backgrounds/{bg}'
        if not os.path.exists(path):
            print(f"{bg} not found!")
            continue
            
        print(f"Cropping {bg}...")
        img = Image.open(path)
        w, h = img.size
        # Crop top half
        cropped = img.crop((0, 0, w, h // 2))
        cropped.save(path, "PNG")
        print(f"Cropped {bg} successfully!")

process_ninja()
process_backgrounds()
